overlord/snapstate/snapstate.go

```diff
@@ -1200,6 +1200,92 @@ func UpdateMany(ctx context.Context, st *state.State, names []string, userID int
 // consider.
 type updateFilter func(*snap.Info, *SnapState) bool
 
+func rearrangeBaseKernelForSingleReboot(kernelTs, bootBaseTs *state.TaskSet) error {
+	var findTaskOfKind = func(ts *state.TaskSet, kind string) (*state.Task, int) {
+		for idx, tsk := range ts.Tasks() {
+			if tsk.Kind() == kind {
+				return tsk, idx
+			}
+		}
+		return nil, -1
+	}
+
+	haveBase, haveKernel := bootBaseTs != nil, kernelTs != nil
+	if !haveBase && !haveKernel {
+		return nil
+	}
+	if haveBase != haveKernel {
+		// have one but not the other
+		var linkSnap *state.Task
+		if haveKernel {
+			// only kernel in this update
+			linkSnap, _ = findTaskOfKind(kernelTs, "link-snap")
+		} else {
+			// only boot base in this update
+			linkSnap, _ = findTaskOfKind(bootBaseTs, "link-snap")
+		}
+		if linkSnap != nil {
+			// we're only updating the base or the kernel, so the link-snap
+			// task can request a reboot
+			linkSnap.Set("can-reboot", true)
```

Comment:
isn't it simpler to keep  can-reboot missing to mean reboot, so we don't have to do this? and we have to set can-reboot only if we have more than rebooting task?